shell/platform/android/io/flutter/plugin/platform/PlatformPlugin.java

```diff
@@ -218,14 +221,74 @@ private void setSystemChromeEnabledSystemUIOverlays(JSONArray overlays) throws J
         updateSystemUiOverlays();
     }
 
-    private void updateSystemUiOverlays() {
+    private void updateSystemUiOverlays(){
         mActivity.getWindow().getDecorView().setSystemUiVisibility(mEnabledOverlays);
+        if (mCurrentTheme != null) {
+            setSystemChromeSystemUIOverlayStyle(mCurrentTheme);
+        }
     }
 
-    private void setSystemChromeSystemUIOverlayStyle(String style) {
+    private void setSystemChromeSystemUIOverlayStyle(JSONObject message) {
         // You can change the navigation bar color (including translucent colors)
-        // in Android, but you can't change the color of the navigation buttons,
-        // so LIGHT vs DARK effectively isn't supported in Android.
+        // in Android, but you can't change the color of the navigation buttons until Android O.
+        // LIGHT vs DARK effectively isn't supported until then.
+
+        // Build.VERSION_CODES.O
+        if (Build.VERSION.SDK_INT < 26) {
+            return;
+        }
+        Window window = mActivity.getWindow();
+        View view = window.getDecorView();
+        int flags = view.getSystemUiVisibility();
+        try {
+            if (!message.isNull("navigationBarColor")) {
+                window.setNavigationBarColor(message.getInt("navigationBarColor"));
+            }
+            if (!message.isNull("navigationDividerColor")) {
+                // Not availible until Android P.
+                // window.setNavigationBarDividerColor(navigationDividerColor);
+            }
+            if (!message.isNull("statusBarColor")) {
+                window.setStatusBarColor(message.getInt("statusBarColor"));
+            }
+            if (!message.isNull("navigationIconTheme")) {
+                String navigationIconTheme = message.getString("navigationIconTheme");
+                switch (navigationIconTheme) {
+                    case "SystemChromeTheme.dark":
+                        //View.SYSTEM_UI_FLAG_LIGHT_NAVIGATION_BAR;
+                        if ((flags & 0x10) == 0) {
```

Comment:
This test and the similar tests below don't seem necessary?

Reply:
My bit-fu is weak, but I believe this is necessary because we currently can set the same system chrome flag multiple times.  If it is already on/off we don't want to toggle it again.

Reply:
But `x |= 0x10` toggle the bit, it just ensures that it's set. `0x10 | 0x10 == 0 | 0x10`